MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -17,7 +17,10 @@ open class RouteProgress: NSObject {
      */
     @objc public var legIndex: Int {
         didSet {
-            assert(legIndex >= 0 && legIndex < route.legs.endIndex)
+            guard legIndex >= 0 && legIndex < route.legs.endIndex else {
+                print("legIndex - \(legIndex) out of range.")
```

Comment:
In addition to @JThramer’s comments, it should be emphasized that `print()` is no substitute for proper error handling. If not for a few existing occurrences that allow certain features to be optional, I’d be in favor of a linter rule against `print()`.